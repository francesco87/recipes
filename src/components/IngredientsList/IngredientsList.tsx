import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import * as React from 'react';

type Item = [string, string];

type Props = {
  ingredients: Item[];
  quantities: Item[];
};

const IngredientsListComponent = ({ ingredients, quantities }: Props) => {
  return (
    <Box sx={{ marginBottom: 3 }}>
      <Typography variant="h6" sx={{ marginBottom: 2 }}>
        INGREDIENTS:
      </Typography>
      <Grid container spacing={1}>
        {ingredients.map((ingredient, index) => (
          <Grid size={6} key={index}>
            <Typography variant="body2" gutterBottom>
              {ingredient[1]}: {quantities[index][1]}
            </Typography>
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};

const IngredientsList = React.memo(IngredientsListComponent);

export default IngredientsList;
