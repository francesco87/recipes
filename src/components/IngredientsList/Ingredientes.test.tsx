import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import IngredientsList from './IngredientsList';

const mockIngredients: [string, string][] = [
  ['id1', 'Tomato'],
  ['id2', 'Onion'],
];

const mockQuantities: [string, string][] = [
  ['q1', '2 pcs'],
  ['q2', '1 pc'],
];

describe('IngredientsList Component', () => {
  it('Render IngredientsList component', () => {
    render(<IngredientsList ingredients={mockIngredients} quantities={mockQuantities} />);
    mockIngredients.forEach((ingredient, index) => {
      const text = `${ingredient[1]}: ${mockQuantities[index][1]}`;
      expect(screen.getByText(text)).toBeInTheDocument();
    });
  });

  it('IngredientsList Snapshot', () => {
    const { container } = render(
      <IngredientsList ingredients={mockIngredients} quantities={mockQuantities} />
    );
    expect(container).toMatchSnapshot();
  });
});
