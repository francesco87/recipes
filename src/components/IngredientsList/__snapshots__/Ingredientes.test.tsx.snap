// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IngredientsList Component > IngredientsList Snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-i3pbo"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1hd3wng-MuiTypography-root"
    >
      INGREDIENTS:
    </h6>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-1 css-1yvsbrk-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-1ml5fja-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ytim1c-MuiTypography-root"
        >
          Tomato
          : 
          2 pcs
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-6 css-1ml5fja-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ytim1c-MuiTypography-root"
        >
          Onion
          : 
          1 pc
        </p>
      </div>
    </div>
  </div>
</div>
`;
